.github/workflows/codeql.yml

```diff
@@ -0,0 +1,34 @@
+name: "Code Scanning - Action"
+
+on:
+  push:
+  schedule:
+    - cron: '0 0 * * 0'
+
+jobs:
+  CodeQL-Build:
+
+    # CodeQL runs on ubuntu-latest, windows-latest, and macos-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
```

Comment:
I enabled the build on three platforms because source files in a native node module may be tied to a specific platform, and hooking the compiler only on one platform won't cover every source file in a project.